### PR TITLE
Generalize `_eventloop.async_context` into `_runstate.ndb_context`.

### PR DESCRIPTION
**Pull request by chrisrossi**
The state managed by `_runstate` will eventually manage all of the runtime
state that falls under the aegis of `context.Context` and `tasklets._State`
in the legacy code.

### CODE REVIEW

File: ndb/src/google/cloud/ndb/_eventloop.py [hunk 1]
```diff
@@ -17,16 +17,12 @@
 This should handle both asynchronous ``ndb`` objects and arbitrary callbacks.
 """
 import collections
-import contextlib
-import threading
 import time
 
-from google.cloud.ndb import exceptions
+from google.cloud.ndb import _runstate as runstate
```
[the remark this answers is not in the document]
Comment (chrisrossi) [reply ndb/src/google/cloud/ndb/_eventloop.py]: No, I just thought it looked a little better to get rid of the underscore. The module itself is private, so it should be safe. I'm not wed to it, though. I can change it.

File: ndb/src/google/cloud/ndb/_eventloop.py [hunk 2]
```diff
@@ -265,104 +261,32 @@ def run(self):
...
     """
-    loop = contexts.current()
-    if loop:
-        return loop
+    state = runstate.current()
 
-    raise exceptions.AsyncContextError()
+    # Be lazy and avoid circular dependency with _runstate
+    if state.eventloop is None:
+        state.eventloop = EventLoop()
```
[the remark this answers is not in the document]
Comment (chrisrossi) [reply ndb/src/google/cloud/ndb/_eventloop.py]: It's in a threadlocal.

File: ndb/src/google/cloud/ndb/_eventloop.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhermes) [reply ndb/src/google/cloud/ndb/_eventloop.py]: I prefer the non-alias version if you're OK with it.

File: ndb/src/google/cloud/ndb/_eventloop.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dhermes) [reply ndb/src/google/cloud/ndb/_eventloop.py]: I'll put on the dunce cap?

[the remark this answers is not in the document]
Comment (chrisrossi) [reply ndb/src/google/cloud/ndb/_eventloop.py]: I'll need it back in a few minutes, but, ok.

**Comment by chrisrossi** [comment]
@dhermes I think I've addressed everything. PTAL.

**Pull request merged**